### PR TITLE
XOR-252 [FEAT] Ensure adding select all option for multiple checkbox form field

### PR DESCRIPTION
**Pull request by NikitaJadhavXoriant**
**Product areas affected**

fliplet-widget-form-builder

**What does this PR do?**
added _Select All_ option in checkbox form field
added via JS API as well, by passing _selectAll_ as true

**JIRA ticket**
[click here](https://weboo.atlassian.net/browse/XOR-252)

**Result**

https://user-images.githubusercontent.com/108272606/179501638-bbbd8d17-8a92-413c-b5ca-c89356468f66.mp4

**Checklist**
 None

**Testing instructions**
None

**Deployment instructions**
None

Author concerns
None

### CODE REVIEW

**Review by yogeshmsharma-xoriant — APPROVED** [review]
Looks ok to me

File: translation.json [hunk 1]
```diff
@@ -58,6 +58,9 @@
         "defaultOptions": {
           "option1": "Option 1",
           "option2": "Option 2"
+        },
+        "isSelectAll": {
```
Comment (squallstar) [inline translation.json:62]: This is incorrect, see https://github.com/Fliplet/fliplet-widget-form-builder/pull/451/files#r923398490

File: templates/configurations/form.interface.hbs [hunk 2]
```diff
@@ -66,6 +66,15 @@
       </div>
     </div>
 
+    <div v-if='_componentsWithSelectAll.includes(_componentName)' class="col-xs-12 form-group">
+      <div class="checkbox checkbox-icon">
+        <input type="checkbox" id="selectAll" name="selectAll" v-model="selectAll" v-on:input="required = false">
```
Comment (squallstar) [inline templates/configurations/form.interface.hbs:71]: Comment not addressed https://github.com/Fliplet/fliplet-widget-form-builder/pull/451/files#r923305925

File: templates/components/checkbox.build.hbs [hunk 3]
```diff
@@ -1,6 +1,29 @@
...
+      >
+      <label v-on:click="checkAllHandler()">
+        <span
+          class="check focus-outline"
+          tabindex="0"
+          v-on:keydown.space.prevent="checkAllHandler()"
+          >
+            <i class="fa fa-check"></i>
+        </span>
+        <span class="option-item">\{{isSelectAll[0].label}}</span>
```
Comment (squallstar) [inline templates/components/checkbox.build.hbs:22]: Comment not addressed https://github.com/Fliplet/fliplet-widget-form-builder/pull/451/files#r923399173

File: templates/components/checkbox.build.hbs [hunk 4]
```diff
@@ -1,6 +1,29 @@
...
+    <div class="checkbox checkbox-icon">
+      <input
+        type="checkbox"
+        :id="name + '-' + 'selectAll'"
+        :name="name"
+        v-model="checkValue"
+        value="Select All"
+        tabindex="-1"
+      >
+      <label v-on:click="checkAllHandler()">
```
Comment (squallstar) [inline templates/components/checkbox.build.hbs:14]: Comment not addressed https://github.com/Fliplet/fliplet-widget-form-builder/pull/451/files

**Review by squallstar — CHANGES_REQUESTED** [review]
@NikitaJadhavXoriant you didn't address any of @tonytlwu's comments in https://github.com/Fliplet/fliplet-widget-form-builder/pull/451/files

**Comment by NikitaJadhavXoriant** [comment]
> @NikitaJadhavXoriant you didn't address any of @tonytlwu's comments in https://github.com/Fliplet/fliplet-widget-form-builder/pull/451/files


Hi Nick, Sorry for misunderstanding. I was waiting for readonly changes to be done after mergning of XOR-156, so Tony's suggested changes was on my local machine, which was not there in PR which you reviewed. But now I have pushed all Tony's suggested changes. So you will get all those changes in PR. Please review.

**Comment by squallstar** [comment]
@tonytlwu this is for you to review 🙏🏻

**Review by tonytlwu — CHANGES_REQUESTED** [review]
Does not meet coding standards

File: js/components/checkbox.js [hunk 5]
```diff
@@ -25,6 +25,63 @@ Fliplet.FormBuilder.field('checkbox', {
...
+      }
+    },
+    selectedAll: {
+      handler: function(val) {
+        if (!this.addSelectAll) {
+          return;
+        }
+
+        var $vm = this;
+
```
Comment (amolsuryawanshi47) [inline js/components/checkbox.js:68]: can you please remove new blank line

Comment (NikitaJadhavXoriant) [reply js/components/checkbox.js]: Done

**Review by amolsuryawanshi47 — DISMISSED** [review]
Changes Looks ok to me

**Review by tonytlwu — CHANGES_REQUESTED** [review]

**Review by amolsuryawanshi47 — APPROVED** [review]
Changes Looks ok to me

**Review by tonytlwu — APPROVED** [review]

**Review by squallstar — APPROVED** [review]

**Pull request merged**